src/main/java/picard/sam/BamErrorMetric/ErrorMetrics.java

```diff
@@ -0,0 +1,169 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package picard.sam.BamErrorMetric;
+
+import htsjdk.samtools.util.QualityUtil;
+import picard.analysis.MergeableMetricBase;
+
+/**
+ *
+ * Base class and implementations for the metrics files required to emit the results of the calculations
+ * in the classes dervied from {@link BaseErrorCalculation} and {@link BaseErrorAggregation}
+ *
+ * @author Yossi Farjoun
+ *
+ */
+public class ErrorMetrics {
+    public static void setPriorError(double priorError) {
+        PRIOR_ERROR = priorError;
+    }
+
+    static double PRIOR_ERROR = 0.001;
+
+    public static class ErrorMetricBase extends MergeableMetricBase {
+        /** The value of the covariate define the bases included in this metric */
+        @MergingIsManual
+        public String COVARIATE;
+        /** The total number of bases included in the calculation of this metric */
+        @MergeByAdding
+        public long TOTAL_BASES;
+
+        public ErrorMetricBase(final String covariate, final long totalBases) {
+            this.TOTAL_BASES = totalBases;
+            this.COVARIATE = covariate;
+        }
+
+        // required to enable reading metric from a file.
+        public ErrorMetricBase() {}
+
+        /** compute a qscore given the number of errors and the total number of bases.
+         * Uses a false count of 1 int the numerator and 1/PRIOR_ERROR in the denominator.*/
+        protected int computeQScore(final long numberOfErrors) {
+            return computeQScore(numberOfErrors, TOTAL_BASES);
+        }
+
+
+        /** compute a qscore given the number of errors and the total number of bases.
+         * Uses a false count of 1 int the denominator and 1 in the numerator.*/
+        protected int computeQScore(final long numberOfErrors, final long nTotalBases) {
+            return QualityUtil.getPhredScoreFromErrorProbability(
+                    (numberOfErrors + PRIOR_ERROR) / (nTotalBases + 1.0D));
+        }
+    }
+
+    /** An error metric for the errors invovling bases in the overlapping region of a read-pair.
+     * The resulting metric includes error rate information which can be assigned to the reading
+     * of the molecular insert {@link #DISAGREES_WITH_REF_AND_MATE_ONLY_Q}, error rate which can be
+     * assigned to events that occured to to the molecular insert before it was loaded onto the
+     * flowcell/sequencer {@link #DISAGREES_WITH_REFERENCE_ONLY_Q}, and an error rate which
+     * cannot be explained nicely {@link #THREE_WAYS_DISAGREEMENT_ONLY_Q}.
+     *
+     */
+    public static class OverlappingErrorMetric extends ErrorMetricBase {
+
+        /** The number of bases for which an overlapping base from the mate read was found*/
```

Comment:
nitpick: space after found

Reply:
How about `The number of bases that had a corresponding overlapping base from their mate`